CONTRIBUTING.md

```diff
@@ -0,0 +1,18 @@
+# Developer guide
+
+Development of the Circuit Knitting Toolbox takes place [on GitHub](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox).  The [Contributing to Qiskit](https://qiskit.org/documentation/contributing_to_qiskit.html) guide may serve as a useful starting point, as the toolbox builds on [Qiskit] and is part of the [Qiskit Ecosystem].
+
+The toolbox is written in [Python] and uses [tox] as a testing framework.  A description of the available `tox` test environments is located at [`test/README.md`](test/README.md).  These environments are used in the CI workflows, which are described at [`.github/workflows/README.md`](.github/workflows/README.md).
+
+Project configuration, including information about dependencies, is stored in [`pyproject.toml`](pyproject.toml).
+
+We use [Sphinx] for documentation and [reno] for release notes.
+
+We require 100% coverage in all new code.  In rare cases where it is not possible to test a code block, we mark it with ``# pragma: no cover`` so that the ``coverage`` tests will pass.
```

Comment:
We might make a comment about our docstring styleand mention some choices we made there (e.g. no hash marks on args in docstrings). It's possible for users to submit code that passes CI but has poor formed docstrings, so it might be worth mentioning. Up to you on this though